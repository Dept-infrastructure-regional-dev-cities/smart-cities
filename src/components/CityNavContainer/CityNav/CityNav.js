import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import style from './CityNav.scss';
import { NO_CITY } from '../../../constants';

const CityNav = props => (
  <div>
    <NavLink
      className={style.link}
      activeClassName={style.link__active}
      to={`/${NO_CITY}/${props.categoryId}`}
    >All Cities</NavLink>

    {props.cities.map(city => (
      <NavLink
        className={style.link}
        activeClassName={style.link__active}
        key={city.id}
        to={`/${city.id}/${props.categoryId}`}
      >{city.name}</NavLink>
    ))}
  </div>
);

CityNav.propTypes = {
  categoryId: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired,
};

export default CityNav;
