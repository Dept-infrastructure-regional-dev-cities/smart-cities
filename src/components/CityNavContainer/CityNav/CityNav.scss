@import '~style/variables';

.link {
  display: block;
  color: white;
  padding: 10px;
  text-decoration: none;
}

.link__active {
  background: white;
  color: teal;
}
