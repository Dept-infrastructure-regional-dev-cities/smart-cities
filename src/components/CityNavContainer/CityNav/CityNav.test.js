import React from 'react';
import { shallow } from 'enzyme';
import CityNav from './CityNav';

const defaultProps = {
  categoryId: 'mock',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
    },
    {
      id: 'sydney',
      name: 'Sydney',
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityNav {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
