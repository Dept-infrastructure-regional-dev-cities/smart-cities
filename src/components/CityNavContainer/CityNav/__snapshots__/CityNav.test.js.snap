// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/all-cities/mock\\" ariaCurrent=\\"true\\">
    All Cities
  </NavLink>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/perth/mock\\" ariaCurrent=\\"true\\">
    Perth
  </NavLink>
  <NavLink className=\\"link\\" activeClassName=\\"link__active\\" to=\\"/sydney/mock\\" ariaCurrent=\\"true\\">
    Sydney
  </NavLink>
</div>"
`;
